### PR TITLE
[MM-5881] Add support for searching by position in user lists

### PR DESCRIPTION
**Pull request by mbecca**
#### Summary

Add support for searching by position in user lists

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-5881

### CODE REVIEW

**Comment by enahum** [comment]
This filter seems to be applied only at the client level.

Should this be added to mobile as well?

**Comment by mbecca** [comment]
> 
> 
> This filter seems to be applied only at the client level, ticket seems to about applying this at the server level.
> 
> Should this be added to mobile as well?

I had not realized that it was from the server, also the changes are on the client side. At the moment I did not make the changes for mobile

File: src/utils/user_utils.ts [hunk 1]
```diff
@@ -150,6 +150,7 @@ export function filterProfilesMatchingTerm(users: Array<UserProfile>, term: stri
         const full = first + ' ' + last;
         profileSuggestions.push(first, last, full);
         profileSuggestions.push((user.nickname || '').toLowerCase());
+        profileSuggestions.push((user.position || '').toLowerCase());
         const email = (user.email || '').toLowerCase();
         profileSuggestions.push(email);
         profileSuggestions.push((user.nickname || '').toLowerCase());
```
Comment (Willyfrog) [inline src/utils/user_utils.ts:156]: not part of your PR, but I wonder why are we adding twice the nickname. @mgdelacroix any ideas?

**Comment by enahum** [comment]
@wiersgallak are we ok with this working only on one client qnd not the other?

File: src/utils/user_utils.ts (hunk 1 above)
Comment (mgdelacroix) [reply src/utils/user_utils.ts:156]: Good catch! It's definitely not necessary, and as the function uses `some`, it does not alter the results, probably why it went unnoticed.

Addressed in https://github.com/mattermost/mattermost-redux/pull/1123

Comment (Willyfrog) [reply src/utils/user_utils.ts:156]: thanks!

**Comment by wiersgallak** [comment]
@enahum I created a ticket for the mobile support: https://mattermost.atlassian.net/browse/MM-24309

@mbecca please let us know if you are interested in adding support for that client as well.  

Thanks!

**Comment by mbecca** [comment]
> 
> 
> @enahum I created a ticket for the mobile support: https://mattermost.atlassian.net/browse/MM-24309
> 
> @mbecca please let us know if you are interested in adding support for that client as well.
> 
> Thanks!

Yes, I can do it !! (in the near future)

Thank you all!

Regards, M

**Pull request merged**